src/UiPath.Workflow.Runtime/ParallelTrackingExtensions.cs

```diff
@@ -9,23 +9,48 @@
 /// </summary>
 public static class ParallelTrackingExtensions
 {
-    public const string BranchIdPropertyName = "__ParallelBranchId";
+    private const string BranchIdPropertyName = "__ParallelBranchId";
 
     public static ActivityInstance MarkNewParallelBranch(this ActivityInstance instance)
     {
-        var properties = new ExecutionProperties(null, instance, instance.PropertyManager);
-        var parentId = properties.Find(BranchIdPropertyName) as string;
-        properties.Add(BranchIdPropertyName, $"{parentId}.{Guid.NewGuid():N}".Trim('.'), true, false);
-
+        var parentId = instance.GetParallelBranchId();
+        instance.SetParallelBranchId(GenerateChildParallelBranchId(parentId));
         return instance;
     }
 
-    public static string GetCurrentParallelBranchId(this ActivityInstance instance) =>
-        instance.PropertyManager?.GetPropertyAtCurrentScope(BranchIdPropertyName) as string;
-
-    public static ActivityInstance MarkNewParallelBranch(this ActivityContext context) =>
-         context.CurrentInstance?.MarkNewParallelBranch();
+    public static string GetParallelBranchId(this ActivityInstance instance) =>

```

Comment:
the same goes for GetCurrentParallelBranchId,

Reply:
?